packages/library/src/commonMain/kotlin/io/realm/RealmConfiguration.kt

```diff
@@ -22,53 +22,143 @@ import io.realm.internal.RealmObjectCompanion
+     * using the [RealmConfiguration] constructor.
+     */
     class Builder(
         var path: String? = null, // Full path for Realm (directory + name)
-        var name: String = "default", // Optional Realm name (default is 'default')
-        vararg var schema: KClass<out RealmObject>
+        var name: String = Realm.DEFAULT_FILE_NAME, // Optional Realm name (default is 'default')
+        var schema: Set<KClass<out RealmObject>> = setOf()
     ) {
+
+        private var logLevel: LogLevel = LogLevel.WARN
+        private var removeSystemLogger: Boolean = false
+        private var customLoggers: List<RealmLogger> = listOf()
+
         fun path(path: String) = apply { this.path = path }
         fun name(name: String) = apply { this.name = name }
-        fun schema(vararg classes: KClass<out RealmObject>) = apply { this.schema = classes }
+        fun schema(classes: Set<KClass<out RealmObject>>) = apply { this.schema = classes }
+        fun schema(vararg classes: KClass<out RealmObject>) = apply { this.schema = setOf(*classes) }
+
+        /**
+         * Configure the [LogLevel] for which all log events of equal or higher priority will be reported.
+         *
+         * @param level Minimum log level to report.
+         */
+        fun logLevel(level: LogLevel) = apply { this.logLevel = level }
+
+        /**
+         * Removes the default system logger from being installed. If no [RealmConfiguration.Builder.customLoggers] have
+         * been configured, not log events will be reported, regardless of the configured [RealmConfiguration.Builder.logLevel].
+         */
+        fun removeSystemLogger() = apply { this.removeSystemLogger = true }
+
+        /**
+         * Install custom loggers to handle log events. Only log events at equal or higher priority than [RealmConfiguration.Builder.logLevel]
+         * will be sent the the loggers. A default system logger is installed by default that will redirect to the common
+         * logging framework on the platform, i.e. LogCat on Android and NSLog on iOS.
+         */
+        fun customLoggers(loggers: List<RealmLogger>) = apply { this.customLoggers = loggers }
 
         fun build(): RealmConfiguration {
             @Suppress("SpreadOperator")
-            return RealmConfiguration(path, name, setOf(*schema))
+            return RealmConfiguration(path, name, schema)
         }
 
-        // Called from compiler plugin
-        internal fun build(companionMap: Map<KClass<*>, RealmObjectCompanion>): RealmConfiguration {
-            return RealmConfiguration(path, name, companionMap)
+        // Called from the compiler plugin
+        internal fun build(companionMap: Map<KClass<out RealmObject>, RealmObjectCompanion>): RealmConfiguration {
+            return RealmConfiguration(
+                companionMap,
+                path,
+                name,
+                schema,
+                LogConfiguration(logLevel, removeSystemLogger, customLoggers)
```

Comment:
I would probably just have initialized the set of loggers with the system logger dependent on the configuration instead of propagating the knowledge of that concept around.